frontend/src/scenes/dashboard/dashboardLogic.tsx

```diff
@@ -1029,14 +1046,23 @@ export const dashboardLogic = kea<dashboardLogicType>([
                 } catch (e: any) {
                     console.error(e)
                     if (isBreakpoint(e)) {
-                        breakpointTriggered = true
+                        cancelled = true
+                    } else if (e.name === 'AbortError' || e.message?.name === 'AbortError') {
+                        if (!cancelled) {
+                            // cancel all insight requests for this query in one go
+                            actions.abortQuery({ queryId: dashboardQueryId })
```

Comment:
Q: Why do we kill something after it's been already KILLed?

Reply:
So, first the AbortController kills the API call that this try/catch is wrapping. That causes this `AbortError` exception. We catch that and use it as the signal to call our cancel API to tell CH to try and cancel it.

We can't listen directly to the `AbortController` (which is a shame)